trees.c

```diff
@@ -599,6 +599,8 @@ void ZLIB_INTERNAL zng_tr_stored_block(deflate_state *s, char *buf, unsigned lon
     sent_bits_add(s, 32);
     if (stored_len)
         memcpy(s->pending_buf + s->pending, (unsigned char *)buf, stored_len);
+    else
+        s->block_open = 0;
     s->pending += stored_len;
```

Comment:
If stored_len is 0 above, this line is no-op, maybe should be moved inside if after memcpy().

Reply:
That is intentional. It should only close the block opened by _zng_tr_emit_tree_ if the length is zero. Apparently _Z_SYNC_FLUSH_ writes a stored block of zero length to help inflate. What was previously happening was this zero length block was being written, however since the block was still opened, _deflate_quick_  (when used in switchlevels) would not emit the beginning of the block when it started deflate.

Reply:
It was only an issue that would be seen in conjunction with `block_open` which uses the flag.

Reply:
However, next three lines after the if else statement can be moved into first conditional of if statement.

Reply:
I meant the line "s->pending += stored_len;" which was last line before the comment... If lines further down should also moved inside the first part of "if" block, then you can do it before I review again...

Reply:
I can also move the setting of _block_open_ to 1 or 0 back to only _deflate_quick_. What do you think?

Reply:
Considering deflate_stored does some weird stuff with empty blocks, I rather see all the modifications in deflate_quick.

Reply:
I made the changes. Thanks for the reivew. I think this looks better now.